### DISCUSSION TITLE
No pacts or verifications have been published Error

### DISCUSSION THREAD

**Discussion by indrabasak** [pact-foundation/pact-js#1064] [Q&A]
I am a pact newbie still trying to figure out how it works. I have a provider and consumer GitHub project. I have followed the same style as the example projects. My consumer project was able to publish the pact successfully but failed at the `can-i-deploy` phase with the following message,

```
There is no verified pact between version 59f6241d950e5cbe1fcf9ecf520925a3b9d6985a of consumer-sqs-pact-example and a version of provider-sqs-pact-example currently in production (no version is currently recorded as deployed/released in this environment)
```
The provider project also failed at the `can-i-deploy` phase with the following message,

```
No pacts or verifications have been published for version 2151cbeeb7f529643150c2d57bf5d6376cb5235c of provider-sqs-pact-example
```

You can find my projects hers:

- [Provider](https://github.com/indrabasak/provider-sqs-pact-example)
-  [Consumer](https://github.com/indrabasak/consumer-sqs-pact-example)

**Comment by mefellows** [comment]
> There is no verified pact between version 59f6241d950e5cbe1fcf9ecf520925a3b9d6985a of consumer-sqs-pact-example and a version of provider-sqs-pact-example currently in production (no version is currently recorded as deployed/released in this environment)

This message says a provider that satisfies the contract is not deployed to `production`, and it is not safe to release.

> No pacts or verifications have been published for version 2151cbeeb7f529643150c2d57bf5d6376cb5235c of provider-sqs-pact-example

This message says "I've never seen that version of the provider before, so I don't know if it's compatible or not". My guess is the version of your provider that verifies the contract is not the same version as the one you're trying to deploy.

I suggest going through this workshop which should clarify the workflow steps/challenges you're having: https://docs.pactflow.io/docs/workshops/ci-cd
